crates/mpc-protocols/src/mpc_service.rs

```diff
@@ -0,0 +1,141 @@
+use crate::mpc_events::{CreatedProofMPCEvent, MPCEvent};
+use log::{debug, info};
+use pera_types::base_types::ObjectID;
+use pera_types::event::Event;
+use std::cell::RefCell;
+use std::collections::{HashMap, VecDeque};
+use tokio::sync::{mpsc, Mutex, RwLock};
+
+// TODO (#253): Make this configurable from a config file
+const MAX_ACTIVE_MPC_INSTANCES: usize = 100;
+
+pub enum MPCInput {
+    InitEvent(CreatedProofMPCEvent),
+    Message,
+}
+
+struct MPCInstance {
+    status: MPCSessionStatus,
+    /// The channel to send message to this instance
+    input_receiver: Option<mpsc::Sender<MPCInput>>,
+    pending_messages: Vec<MPCInput>,
+}
+
+impl MPCInstance {
+    fn set_active(&mut self) {
+        self.status = MPCSessionStatus::Active;
+        // TODO (#256): Replace hard coded 100 with the number of validators times 10
```

Comment:
why times 10?

Reply:
Seemed reasonable for me at the time, the entire channels logic has been removed in a later PR anyway as it's not needed